packages/ariakit-utils/src/__tests__/dom-test.ts

```diff
@@ -0,0 +1,601 @@
+  expect(getActiveElement(input)).toEqual(input);
+
+  // Test IE11 Case
+  const activeElementMock = jest
+    .spyOn(document, "activeElement", "get")
+    .mockReturnValue(null);
+
+  expect(getActiveElement(undefined)).toEqual(null);
+  expect(getActiveElement(null)).toEqual(null);
+  expect(getActiveElement(node)).toEqual(null);
+  expect(getActiveElement(input)).toEqual(null);
+
+  activeElementMock.mockRestore();
+
+  // TODO: Not sure how to test the iframe case
+  //  One option might be mocking isFrame and activeElement
+
+  // activeDecendant = true
+  const input2 = document.getElementById("cb1-edit");
+  input2?.focus();
+
+  const focusedLi = document.getElementById("cb1-opt6");
+
+  expect(getActiveElement(undefined, true)).toEqual(focusedLi);
+  expect(getActiveElement(null, true)).toEqual(focusedLi);
+  expect(getActiveElement(node, true)).toEqual(focusedLi);
+  expect(getActiveElement(input, true)).toEqual(focusedLi);
+  expect(getActiveElement(input2, true)).toEqual(focusedLi);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("contains", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML =
+    '<div id="testNode"><div id="innerTestNode" /></div>';
+  const node = document.getElementById("testNode") as HTMLElement;
+  const innerNode = document.getElementById("innerTestNode") as HTMLElement;
+
+  expect(contains(node, node)).toEqual(true);
+  expect(contains(node, innerNode)).toEqual(true);
+  expect(contains(innerNode, node)).toEqual(false);
+
+  // Operates the same as default `conatins`
+  expect(contains(node, node)).toEqual(node.contains(node));
+  expect(contains(node, innerNode)).toEqual(node.contains(innerNode));
+  expect(contains(innerNode, node)).toEqual(innerNode.contains(node));
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("isFrame", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML =
+    '<div id="testNode"><iframe id="testIframe"/></div>';
+  const node = document.getElementById("testNode") as HTMLElement;
+  const iframe = document.getElementById("testIframe") as HTMLIFrameElement;
+
+  expect(isFrame(node)).toEqual(false);
+  expect(isFrame(iframe)).toEqual(true);
+
+  // Reset
+  document.body.innerHTML = initialInnerHTML;
+});
+
+test("isButton", () => {
+  const initialInnerHTML = document.body.innerHTML;
+  document.body.innerHTML = `
+    <div id="testNode">
+      <button id="testButton"/>
+      <input type="button" id="testInputButton"/>
+      <input type="submit" id="testInputSubmit"/>
+      <input type="reset" id="testInputReset"/>
+      <input type="image" id="testInputImage"/>
+      <input type="file" id="testInputFile"/>
+      <input type="color" id="testInputColor"/>
+      <input type="text" id="testInput"/>
+      <div role="button" id="testDivButton"/>
+    </div>
+  `;
+  const node = document.getElementById("testNode") as HTMLElement;
```

Comment:
Since we're using a lot `getElementById` here, I'd create a helper function at the top of this file (just for this file):
```ts
function getById<T extends HTMLElement = HTMLElement>(id: string) {
  return document.getElementById(id) as T;
}

// usage
getById("testNode");
getById<HTMLInputElement>("textInputSubmit");
```